docs/worldgen/biomemodifier.md

```diff
@@ -0,0 +1,648 @@
+    "biomes": "minecraft:plains",
+    // Can either be an id "examplemod:add_carvers_example"
+    // List of ids ["examplemod:add_carvers_example", "minecraft:canyon", ...]
+    // Or a tag "#examplemod:configured_carver_tag"
+    "carvers": "examplemod:add_carvers_example",
+    // See GenerationStep.Carving in code for a list of valid enum names.
+    // Only `air` and `liquid` are available.
+    "step": "air"
+}
+```
+
+### Removing Legacy Carvers
+
+This Biome Modifier type allows removing old caves and ravines from biomes. This CANNOT remove Noise Caves to biomes because Noise Caves are baked into the dimension's Noise Setting system and not actually tied to biomes. The legacy carvers are specifically ravines and old caves.
+
+```json5
+{
+    "type": "neoforge:remove_carvers",
+    // Can either be an id "minecraft:plains"
+    // List of ids ["minecraft:plains", "minecraft:badlands", ...]
+    // Or a tag "#c:is_overworld"
+    "biomes": "#c:is_overworld",
+    // Can either be an id "minecraft:cave"
+    // List of ids ["minecraft:cave", "minecraft:canyon", ...]
+    // Or a tag "#examplemod:configured_carver_tag"
+    "carvers": "minecraft:cave",
+    // Can either be a single step "air"
+    // Or a list ["air", "liquid"]
+    // See GenerationStep.Carving for a list of valid enum names.
+    "steps": [
+        "air",
+        "liquid"
+    ]
+}
+```
+
+### Word of Warning
+
+- Avoid using Biome Modifiers to add vanilla placed features to biomes, as this may cause a feature cycle violation (the game will crash if two biomes have the same two features in their feature lists but in different orders within same GenerationStep). Placed features can be referenced in biome jsons or added via Biome Modifiers, but should not be used in both. Make a new copy of a vanilla Placed Feature is ideal for adding it safely to biomes.
+
+
+- Avoid adding the same placed feature with more than one Biome Modifier, as this can cause feature cycle violations.
+
+### The available values for the Decoration steps
+
+The `step` field in many of these JSONs are referring to GenerationStep.Decoration enum. This enum has the steps listed out in this order which is the same order that the game uses for generating during worldgen:
+
+    raw_generation
+    lakes
+    local_modifications
+    underground_structures
+    surface_structures
+    strongholds
+    underground_ores
+    underground_decoration
+    fluid_springs
+    vegetal_decoration
+    top_layer_modification
+
+
+## Datagenning Biome Modifiers:
+
+A `BiomeModifier` can be with [data generation][datagen] by passing a `RegistrySetBuilder` to `DatapackBuiltinEntriesProvider`. Biome Modifiers are located within `data/<modid>/neoforge/biome_modifier/<path>.json` All Biome Modifiers contain a `type` key that references the id of the `MapCodec` used for the Biome Modifier. All other settings provided by the Biome Modifier are added as additional keys on the root object.
```

Comment:
This section is way too sparse to understand what to do here. How do I create the biome modifier? Do I need to register the map codec? How does this link together?

Reply:
I am not well verse in datagen so some one else would need to fill this out with more info. I took what info your link had to put here

Reply:
Yeah, my comment was because the example biome modifier was referenced up here before it was defined.